lib/omniauth/strategies/saml.rb

```diff
@@ -44,10 +45,6 @@ def request_phase
       end
 
       def callback_phase
-        unless request.params['SAMLResponse']
-          raise OmniAuth::Strategies::SAML::ValidationError.new("SAML response missing")
```

Comment:
@ErinCall: Nice catch!

Since that's the only call to `response_fingerprint`, it's probably safe to check for `request.params['SAMLResponse']` in that method and return `false` or `nil` if is is not present. This will cause the code to raise `OmniAuth::Strategies::SAML::ValidationError` as it would have previous to this PR.

In general, I think this `idp_cert_fingerprint_validator` code needs to be deprecated and eventually removed, but it's possible I don't understand the use case. I think it's a bad idea to have `ominiauth-saml` directly decoding the `SAMLResponse` instead of relying on `ruby-saml` to do so in a consistent way.